Source/WebCore/page/Navigation.cpp

```diff
@@ -396,6 +454,35 @@ static bool documentCanHaveURLRewritten(const RefPtr<Document>& document, const
     return equalIgnoringFragmentIdentifier(documentURL, targetURL);
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#promote-an-upcoming-api-method-tracker-to-ongoing
+void Navigation::promoteUpcomingAPIMethodTracker(const String& destinationKey)
+{
+    ASSERT(!m_ongoingAPIMethodTracker);
+
+    if (!destinationKey.isNull()) {
+        // FIXME: ASSERT(!m_upcomingNonTraverseMethodTracker);
```

Comment:
// FIXME: We ought to be able to assert that m_upcomingNonTraverseMethodTracker is nullopt here.